collect_app/src/main/java/org/odk/collect/android/dao/helpers/InstancesDaoHelper.java

```diff
@@ -44,7 +44,7 @@ public static boolean isInstanceComplete(boolean end) {
         boolean complete = false;
 
         FormController formController = Collect.getInstance().getFormController();
-        if (formController != null) {
+        if (formController != null && formController.getInstancePath() != null) {
```

Comment:
Would be good to update the log message on line 65 as well to make it easier to track down any possible problems.